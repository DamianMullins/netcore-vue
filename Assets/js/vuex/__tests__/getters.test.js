import { getters } from '../';

describe('getters', () => {
    const state = {
        menu: {
            items: [
                { id: 1, name: 'One', price: 1.11 },
                { id: 2, name: 'Two', price: 2.22 }
            ]
        },
        basket: {
            items: [
                { id: 2, quantity: 2 }
            ],
            deliveryFee: 0.5
        }
    };

    describe('basketItems', () => {
        it('gets basket items with correct properties', () => {
            // Arrange
            const { basketItems } = getters;

            // Act
            const result = basketItems(state);

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('subtotal', () => {
        it('gets correct value for subtotal', () => {
            // Arrange
            const { subtotal } = getters;
            const gettersMock = jest.fn();

            gettersMock.basketItems = [{
                id: 2,
                name: 'Two',
                price: 2.22,
                quantity: 2,
            }];

            // Act
            const result = subtotal(state, gettersMock);

            // Assert
            expect(result).toBe(4.44);
        });
    });

    describe('total', () => {
        it('gets correct value for total', () => {
            // Arrange
            const { total } = getters;
            const gettersMock = jest.fn();

            gettersMock.subtotal = 2.22;

            // Act
            const result = total(state, gettersMock);

            // Assert
            expect(result).toBe(2.72);
        });
    });
});
