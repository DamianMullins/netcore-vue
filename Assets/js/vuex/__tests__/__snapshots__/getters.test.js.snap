// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getters basketItems gets basket items with correct properties 1`] = `
Array [
  Object {
    "id": 2,
    "name": "Two",
    "price": 2.22,
    "quantity": 2,
  },
]
`;
